### PR TITLE
mzcompose: Use the "default" workflow by default in CI

### PR DESCRIPTION
**Pull request by quodlibetor**
Add a "default" workflow to almost all mzcompose.py files.

I created *new* workflows in two files:

* test/mzcompose_examples/mzcompose.py
* test/sqllogictest/mzcompose.py

### Motivation

This makes it possible to cd into any mzcompose directory and run `mzcompose run default` and get reasonable behavior.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [n/a] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: ci/plugins/mzcompose/hooks/command
Comment (guswynn) [inline ci/plugins/mzcompose/hooks/command:42]: dont we all love bash syntax?

File: test/sqllogictest/mzcompose.py
Comment (guswynn) [inline test/sqllogictest/mzcompose.py:19]: why forward this one? just to keep the names consistent down below?

**Comment by guswynn** [comment]
This seems good to me, but probably @benesch wants to look at it

one question is: should we make it so mzcompose itself errors if the composition does not have an "default" workflow?

**Review by danhhz — COMMENTED** [review]
also lgtm but @benesch is the real reviewer here

File: test/sqllogictest/mzcompose.py
Comment (quodlibetor) [reply test/sqllogictest/mzcompose.py:19]: Yeah it just felt more clear here since there are multiple equivalently-important workflows. All the other ones where I renamed them the name of the workflow added no value compared to the name of the directory.

I might add doc parsing to mzcompose so that `mzcompose list` can provide explanations, which would eliminate this.

Comment (benesch) [reply test/sqllogictest/mzcompose.py:19]: It already does that! Add a doc comment and it will show up in `mzcompose list`.

Comment (benesch) [reply test/sqllogictest/mzcompose.py:19]: This is better work for a followup, but I'd say we should rename `slt.sh` to `slt-slow.sh`, `slt-fast.sh` to `slt.sh`, and then have one workflow called "default" that takes a `--slow` option.

**Review by benesch — APPROVED** [review]
So good! We should wait for @philip-stoev's sign-off too but this is great.

**Review by philip-stoev — APPROVED** [review]
Thanks, looks good to me!

**Comment by philip-stoev** [comment]
If you could fix any .md files that reference `mzcompose` and that may mention workload names that would be much appreciated. (Please rebase first though)

**Comment by quodlibetor** [comment]
> If you could fix any .md files that reference mzcompose and that may mention workload names

Done, afaict (I checked via `rg mzcompose.*run`).

**Comment by benesch** [comment]
What's up with the "Update debezium-avro test docs to reflect that they are DDL tests" commit? Those tests seem to test a _lot_ more than just DDL.

**Comment by quodlibetor** [comment]
We talked in slack and agreed on the new structure for that commit.

**Pull request merged**
